storage/SystemInfo/CmdBtrfs.cc

```diff
@@ -203,6 +203,18 @@ namespace storage
 	    if (boost::starts_with(entry.path, "<FS_TREE>/"))
 		entry.path.erase(0, strlen("<FS_TREE>/"));
 
+	    string::size_type pos4 = line.find(" uuid ");
+	    if (pos4 == string::npos)
+		ST_THROW(Exception("could not find 'uuid' in 'btrfs subvolume list' output"));
+	    line.substr(pos4 + 6) >> entry.uuid;
```

Comment:
How about `sizeof("somestring") - 1` for all those numbers to get an idea what we are skipping over?

Reply:
Sure.